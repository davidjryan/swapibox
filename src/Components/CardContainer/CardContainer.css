.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  background: #3F0E10;
  margin: 15px;
  overflow: auto; }
