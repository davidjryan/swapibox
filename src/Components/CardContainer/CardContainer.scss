@import "../../variables.scss";

.card-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  background: $body-dark;
  margin: 15px;
  overflow: auto;
}
