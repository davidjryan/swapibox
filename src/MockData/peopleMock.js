module.exports = [

    {
      Name: "Luke Skywalker",
      Species: "Human",
      Language: "Galactic Basic",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "C-3PO",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "200000",
      Homeworld: "Tatooine",
      favorite: false
    },
    {
      Name: "R2-D2",
      Species: "Droid",
      Language: "n/a",
      "Homeworld Population": "4500000000",
      Homeworld: "Naboo",
      favorite: false
    }

];
