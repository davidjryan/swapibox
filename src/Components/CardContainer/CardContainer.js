import React from 'react';
import Card from '../Card/Card'
import './CardContainer.css'
import PropTypes from 'prop-types';

const CardContainer = ({cardData, toggleFav}) => {
  const cardFormat = cardData.map( (item) => {
    return(
      <Card cardInformation={ item } toggleFav={ toggleFav } key={`${item.name} - ${Date.now()}`} />
    )
  })
  return (
    <div className="card-container">
      { cardFormat }
    </div>
  )
}

export default CardContainer
