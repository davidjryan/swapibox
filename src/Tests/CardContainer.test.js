import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from '../Components/CardContainer/CardContainer';
import peopleMock from '../MockData/peopleMock';

describe('CardContainer', () => {

  it('should create an instance of a card container', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<CardContainer
      cardData={peopleMock}
      toggleFav={mockFunc}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
