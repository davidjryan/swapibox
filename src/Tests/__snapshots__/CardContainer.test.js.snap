// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should create an instance of a card container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardData={
        Array [
            Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "Galactic Basic",
              "Name": "Luke Skywalker",
              "Species": "Human",
              "favorite": false,
            },
            Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "n/a",
              "Name": "C-3PO",
              "Species": "Droid",
              "favorite": false,
            },
            Object {
              "Homeworld": "Naboo",
              "Homeworld Population": "4500000000",
              "Language": "n/a",
              "Name": "R2-D2",
              "Species": "Droid",
              "favorite": false,
            },
          ]
    }
    toggleFav={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardInformation={
                    Object {
                              "Homeworld": "Tatooine",
                              "Homeworld Population": "200000",
                              "Language": "Galactic Basic",
                              "Name": "Luke Skywalker",
                              "Species": "Human",
                              "favorite": false,
                            }
          }
          className="card"
          toggleFav={[Function]}
/>,
        <Card
          cardInformation={
                    Object {
                              "Homeworld": "Tatooine",
                              "Homeworld Population": "200000",
                              "Language": "n/a",
                              "Name": "C-3PO",
                              "Species": "Droid",
                              "favorite": false,
                            }
          }
          className="card"
          toggleFav={[Function]}
/>,
        <Card
          cardInformation={
                    Object {
                              "Homeworld": "Naboo",
                              "Homeworld Population": "4500000000",
                              "Language": "n/a",
                              "Name": "R2-D2",
                              "Species": "Droid",
                              "favorite": false,
                            }
          }
          className="card"
          toggleFav={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker - 0",
        "nodeType": "function",
        "props": Object {
          "cardInformation": Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "Galactic Basic",
            "Name": "Luke Skywalker",
            "Species": "Human",
            "favorite": false,
          },
          "className": "card",
          "toggleFav": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "C-3PO - 1",
        "nodeType": "function",
        "props": Object {
          "cardInformation": Object {
            "Homeworld": "Tatooine",
            "Homeworld Population": "200000",
            "Language": "n/a",
            "Name": "C-3PO",
            "Species": "Droid",
            "favorite": false,
          },
          "className": "card",
          "toggleFav": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "R2-D2 - 2",
        "nodeType": "function",
        "props": Object {
          "cardInformation": Object {
            "Homeworld": "Naboo",
            "Homeworld Population": "4500000000",
            "Language": "n/a",
            "Name": "R2-D2",
            "Species": "Droid",
            "favorite": false,
          },
          "className": "card",
          "toggleFav": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardInformation={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Homeworld Population": "200000",
                                    "Language": "Galactic Basic",
                                    "Name": "Luke Skywalker",
                                    "Species": "Human",
                                    "favorite": false,
                                  }
            }
            className="card"
            toggleFav={[Function]}
/>,
          <Card
            cardInformation={
                        Object {
                                    "Homeworld": "Tatooine",
                                    "Homeworld Population": "200000",
                                    "Language": "n/a",
                                    "Name": "C-3PO",
                                    "Species": "Droid",
                                    "favorite": false,
                                  }
            }
            className="card"
            toggleFav={[Function]}
/>,
          <Card
            cardInformation={
                        Object {
                                    "Homeworld": "Naboo",
                                    "Homeworld Population": "4500000000",
                                    "Language": "n/a",
                                    "Name": "R2-D2",
                                    "Species": "Droid",
                                    "favorite": false,
                                  }
            }
            className="card"
            toggleFav={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker - 0",
          "nodeType": "function",
          "props": Object {
            "cardInformation": Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "Galactic Basic",
              "Name": "Luke Skywalker",
              "Species": "Human",
              "favorite": false,
            },
            "className": "card",
            "toggleFav": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "C-3PO - 1",
          "nodeType": "function",
          "props": Object {
            "cardInformation": Object {
              "Homeworld": "Tatooine",
              "Homeworld Population": "200000",
              "Language": "n/a",
              "Name": "C-3PO",
              "Species": "Droid",
              "favorite": false,
            },
            "className": "card",
            "toggleFav": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "R2-D2 - 2",
          "nodeType": "function",
          "props": Object {
            "cardInformation": Object {
              "Homeworld": "Naboo",
              "Homeworld Population": "4500000000",
              "Language": "n/a",
              "Name": "R2-D2",
              "Species": "Droid",
              "favorite": false,
            },
            "className": "card",
            "toggleFav": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
